back-end/utils/top-layer/UserUtil.js

```diff
@@ -362,6 +362,45 @@ const getLengthUserTransactionsHistoryForRedisM5RU = (email, filters) => {
   });
 };
 
+/**
+ * Buy or Sell a single transaction
+ * @param {string} transactionID
+ * @param {number} recentPrice
+ */
+const proceedTransaction = (transactionID, recentPrice) => {
```

Comment:
Khi xử lí xong transaction, anh cũng đừng quên set những attribute của Transaction (có ghi trong schema) lại cho đúng
- isFinished thì set true
- finishedTime thì set giờ hiện tại
- update priceAtTransaction thành price hiện tại
- spendOrGain = priceAtTransaction * quantity (+ hoặc - brokerage). + nếu là buy. - nếu là sell. Nếu type là Buy thì anh nhớ cast cái spendOrGain thành số âm (do nó có nghĩa là anh spend money to buy stock)